schema/encoding/jsonschema.go

```diff
@@ -84,7 +84,7 @@ func SchemaToJSONSchema(s *schema.Schema) string {
 			required = append(required, fmt.Sprintf(`"%s"`, key))
 		}
 		if field.ReadOnly {
-			result += fmt.Sprintf("%q: %t, ", "readonly", field.ReadOnly)
+			result += fmt.Sprintf("%q: %s, ", "readonly", `{ "type": "boolean" }`)
```

Comment:
This does not look better I am afraid. Short story is that JSON-Schema Draft 4 simply does not have a way of encoding this information (officially). However Swagger 2.0 have defined an "extension" for it with a field called "readOnly" (See http://swagger.io/specification/#schemaObject).

See new comment on initial commit for more detail.